src/Authenticator/CookieAuthenticator.php

```diff
@@ -0,0 +1,121 @@
+use Cake\Http\Response;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use RuntimeException;
+
+/**
+ * Cookie Authenticator
+ *
+ * Authenticates an identity based on a cookies data.
+ */
+class CookieAuthenticator extends AbstractAuthenticator implements PersistenceInterface
+{
+
+    /**
+     * {@inheritDoc}
+     */
+    public function __construct(IdentifierCollection $identifiers, array $config = [])
+    {
+        $this->_checkCakeVersion();
+
+        $this->_defaultConfig['cookie'] = [
+            'name' => 'CookieAuth',
+            'expire' => null,
+            'path' => '/',
+            'domain' => '',
+            'secure' => false,
+            'httpOnly' => false,
+            'value' => ''
+        ];
+
+        parent::__construct($identifiers, $config);
+    }
+
+    /**
+     * Checks the CakePHP Version by looking for the cookie implementation
+     *
+     * @return void
+     */
+    protected function _checkCakeVersion()
+    {
+        if (!class_exists('Cake\Http\Cookie\Cookie')) {
+            throw new RuntimeException('You must use the CakePHP `3.next` branch or wait until CakePHP version 3.5 is released to use the CookieAuthenticator');
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function authenticate(ServerRequestInterface $request, ResponseInterface $response)
+    {
+        $userData = $request->getCookie($this->getConfig('cookie.name'));
+
+        if (empty($userData)) {
+            return new Result(null, Result::FAILURE_CREDENTIALS_NOT_FOUND, [
+                'Login credentials not found'
+            ]);
+        }
+
+        $user = $this->identifiers()->identify($userData);
+
+        if (empty($user)) {
+            return new Result(null, Result::FAILURE_IDENTITY_NOT_FOUND, $this->identifiers()->getErrors());
+        }
+
+        return new Result($user, Result::SUCCESS);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function persistIdentity(ServerRequestInterface $request, ResponseInterface $response, $identity)
+    {
+        $cookie = $this->getConfig('cookie');
+        $cookie['value'] = $identity;
+
+        return [
+            'request' => $request,
+            'response' => $response->withCookie(
+                $cookie['name'],
+                $cookie
+            )
+        ];
+    }
+
+    /**
+     * Clears the identity data
+     *
+     * @param \Psr\Http\Message\ServerRequestInterface $request The request object.
+     * @param \Cake\Http\Response $response The response object.
+     * @return array Returns an array containing the request and response object
+     */
+    public function clearIdentity(ServerRequestInterface $request, ResponseInterface $response)
+    {
+        $cookie = $this->getConfig('cookie');
+        $cookie['expire'] = strtotime('-1 year');
+
+        return [
+            'request' => $request,
+            'response' => $response->withCookie(
```

Comment:
@robertpustulka all of this has already been discussed, it just needs to be implemented. :)

> 12:26] 
burzum Hi Mark, any idea how to resolve these two? https://travis-ci.org/cakephp/authentication/jobs/229100440
[12:26]
slackbot APP markstory has snoozed notifications. Send one anyway? Send notification
Only visible to you
[12:26] 
burzum The cause is that the methods use the Interface and the interface doesn't implement these methods.
[12:26] 
But our ServerRequest object does
[12:27] 
If we want to keep this generic the solution would be to check if the instance is a cake ServerRequest and otherwise use whatever the default it
[18:42] 
markstory I would remove the typehint or use the correct one.
[19:09] 
burzum The main question is this: Do we want to make the cookie authenticator to work without cakephp as well?
[19:10] 
If we rely on cakes ServerRequest and Response we'll introduce a dependency we don't have yet
[22:19] 
markstory I am ok with that. Cookies with the base objects is pretty awkward
[23:14] 
burzum So this means you want what I suggest, support for both OR only cake support? :slightly_smiling_face: